server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/shards/TransportCatShardsActionIT.java

```diff
@@ -125,4 +130,101 @@ public void onFailure(Exception e) {
         latch.await();
     }
 
+    public void testListShardsWithClosedAndHiddenIndices() throws InterruptedException, ExecutionException {
```

Comment:
Can we add tests with cluster-state containing
- only closed index
- only hidden index
- combination of hidden and closed
We can repeat the same queries as this test and verify output response

Reply:
Updated in the revision.